### PR TITLE
EOS-26878: add cortx.const in utils for defining constants of service name and components

### PR DESCRIPTION
**Pull request by rahul-tripathi-git**
Signed-off-by: Rahul Tripathi <rahul.tripathi@seagate.com>

# Problem Statement
- Added cortx constants for service names and component names

# Design
-  Added a new subpackage.module -> cortx.utils.cortx.const.py
- To import const class -> `from cortx.utils.cortx import Const` OR `from cortx.utils.cortx.const import Const`

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: N/A
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: N/A
- [x] Confirm Testing was performed with installed RPM [Y/N]:  N/A

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [x] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/cortx/const.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    # Components
+    COMPONENT_UTILS     = "utils"
+    COMPONENT_MOTR      = "motr"
+    COMPONENT_HARE      = "hare"
+    COMPONENT_S3        = "s3"
+    COMPONENT_HA        = "ha"
+    COMPONENT_CSM       = "csm"
+
+    # Services
```
Comment (sachinpunadikar) [inline py-utils/src/utils/cortx/const.py:31]: don't we need SERVICE_MESSAGE_SERVER  = "message_server" ?

Comment (nitin-seagate) [reply py-utils/src/utils/cortx/const.py:31]: @sachinpunadikar If its utils specific then lets add it as:
`SERVICE_UTILS_MSGSERVER = "message_server"`

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
@rahul-tripathi-git Please update the constants in-line with ticket specifications (some new constants added in ticket)

File: py-utils/src/utils/cortx/const.py (hunk 1 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/cortx/const.py]: Added as suggested by @nitin-seagate below. Kindly re-review.
`SERVICE_UTILS_MSGSERVER = "message_server"`

**Comment by rahul-tripathi-git** [comment]
> @rahul-tripathi-git Please update the constants in-line with ticket specifications (some new constants added in ticket)

Done.

File: py-utils/src/utils/cortx/const.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    SERVICE_S3_BGSCHEDULER  = "bgscheduler"
+    SERVICE_CSM_AGENT       = "agent"
+    SERVICE_UTILS_MSGSERVER = "message_server"
+
+    # Deprecated services
+    SERVICE_S3_IO          = "io"
+    SERVICE_S3_AUTH        = "auth"
+    SERVICE_S3_BGCONSUMER  = "bg_consumer"
+    SERVICE_S3_BGPRODUCER  = "bg_producer"
+    SERVICE_S3_OPENLDAP    = "openldap"
```
Comment (vastradparayya) [inline py-utils/src/utils/cortx/const.py:48]: @rahul-tripathi-git  should values be in single as all are single words?

File: py-utils/src/utils/cortx/const.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    COMPONENT_HARE         = "hare"
+    COMPONENT_S3           = "s3"
+    COMPONENT_HA           = "ha"
+    COMPONENT_CSM          = "csm"
+
+    # Active Services
+    SERVICE_MOTR_IO         = "io"
+    SERVICE_MOTR_FSM        = "fsm"
+    SERVICE_HARE_HAX        = "hax"
+    SERVICE_S3_SERVER       = "s3server"
```
Comment (shriya-deshmukh) [inline py-utils/src/utils/cortx/const.py:35]: Can we define it as SERVICE_S3_S3SERVER , As we have followed same format('SERVICE_<COMP-NAME>_<SERVICE-NAME>') for all other keys?

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/utils/cortx/const.py (hunk 2 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/cortx/const.py]: Done.

File: py-utils/src/utils/cortx/const.py (hunk 3 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/cortx/const.py]: @ujjwalpl kindly suggest on above.

**Review by vastradparayya — APPROVED** [review]

File: py-utils/src/utils/cortx/const.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from enum import Enum
+
+class Const(Enum):
+    """Enum class to define constants for components and service names."""
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/cortx/const.py:21]: https://github.com/PyCQA/pydocstyle/issues/141

**Review by nitin-seagate — APPROVED** [review]
LGTM

File: py-utils/src/utils/cortx/const.py (hunk 3 above)
Comment (nitin-seagate) [reply py-utils/src/utils/cortx/const.py]: SERVICE_S3_S3SERVER will make it cumbersome to read...

Comment (nitin-seagate) [reply py-utils/src/utils/cortx/const.py]: SERVICE_S3_SERVER should be fine to understand and read...

**Review by shriya-deshmukh — APPROVED** [review]

**Pull request merged**